nuodb/nuodb-helm-charts
# Fix B&R issues with separate journal directory

**Issue**
This PR fixes issues discussed in https://github.com/nuodb/nuodb-helm-charts/pull/218#issuecomment-859604537.

**Changes**
- Use external journal path when creating the archive metadata using `nuodocker restore archive`. Otherwise, the archive will be created with an empty `journalPath` and the SM won't be provided with `--journal-dir` option. This problem was masking the original issue.
- Move journal content from `stream` backup to external journal directory as the journal is not consumed during engine shutdown.

**Testing**
- add a new database import test with a separate journal directory. The model import archive already has journal content (see https://github.com/nuodb/nuodb-helm-charts/pull/218#issuecomment-859678181).
- insert additional verifications in the existing test for the archive metadata when a separate journal is used
- regression testing
